addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/connector/SerialPortManager.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.internal.device.connector;
+
+import java.util.Enumeration;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+
+/**
+ * Interface for a serial port manager.
+ *
+ * This interface is modeled after the smart home serial transport interface. Because that interface is not yet
```

Comment:
What is missing in the ESH Serial API? I a not aware of any open issues that you might be waiting for.

Reply:
I haven't created any open issues (yet) :smile: I did replace the class implementation with the ESH interface to test it and it misses some methods. I don't know if these method should be added or it it would require the usage to be changed a little. I didn't change much from the original implementation on how the methods of library are used from the original implementation so to not get off track too much I designed most of the code that the injection would be fairly easy and thus if it's as simple as adding the methods to the interface it should be fairly easy to add.